crates/pallet-domains/src/staking.rs

```diff
@@ -2145,32 +2179,526 @@ pub(crate) mod tests {
+                maybe_deposit: None,
+                expected_withdraw: Some(expected_withdraw),
+                expected_nominator_count_reduced_by: 0,
+                storage_fund_change: (true, 0),
+            })
+        });
+    }
+
+    /// Property test for withdrawing all a nominator's excess stake.
+    /// Their balance should be almost the same before and after.
+    #[test]
+    fn prop_withdraw_excess_operator_stake_with_nominator() {
+        prop_test!(&(OPERATOR_STAKE_RANGE, NOMINATOR_STAKE_RANGE,), |(
+            operator_stake,
+            nominator_stake,
+        )| {
+            // Total balances can't overflow: arithmetic overflow error in withdraw_stake test function
+            prop_assert!(
+                [operator_stake, nominator_stake]
+                    .into_iter()
+                    .try_fold(0_u128, |acc, value| acc.checked_add(value))
+                    .is_some()
             );
+
+            let expected_withdraw = (nominator_stake, true);
+
+            let excess_stake = expected_withdraw
+                .0
+                .saturating_sub(<Test as Config>::MinNominatorStake::get());
+
+            prop_assert!(excess_stake > 0, "would cause ZeroWithdraw error");
+
+            withdraw_stake_prop(WithdrawParams {
+                minimum_nominator_stake: <Test as Config>::MinNominatorStake::get(),
+                nominators: vec![(0, operator_stake), (1, nominator_stake)],
+                operator_reward: 0,
+                nominator_id: 1,
+                withdraws: vec![(excess_stake, Ok(()))],
+                maybe_deposit: None,
+                expected_withdraw: Some(expected_withdraw),
+                expected_nominator_count_reduced_by: 1,
+                storage_fund_change: (true, 0),
+            })
         });
     }
 
+    /// Property test for withdrawing an operator's excess stake with a deposit.
+    /// Their balance should be almost the same before and after.
+    #[test]
+    fn prop_withdraw_excess_operator_stake_with_deposit() {
+        prop_test!(&(OPERATOR_STAKE_RANGE, DEPOSIT_RANGE,), |(
+            mut operator_stake,
+            maybe_deposit,
+        )| {
+            operator_stake = operator_stake.saturating_add(maybe_deposit);
+
+            prop_assert!(
+                operator_stake.saturating_sub(maybe_deposit)
+                    >= <Test as Config>::MinOperatorStake::get(),
+                "would cause MinimumOperatorStake error"
+            );
+
+            // Total balances can't overflow: arithmetic overflow error in withdraw_stake test function
+            prop_assert!(
+                [operator_stake, maybe_deposit]
+                    .into_iter()
+                    .try_fold(0_u128, |acc, value| acc.checked_add(value))
+                    .is_some()
+            );
+
+            let expected_withdraw = (
+                STORAGE_FEE_RESERVE
+                    .left_from_one()
+                    .mul_ceil(operator_stake)
```

Comment:
Confused about why we need to account for the `STORAGE_FEE_RESERVE` here since it is already taken care of in:
https://github.com/autonomys/subspace/blob/715c3c39cc9e14c09afc625542966f8f94efea56/crates/pallet-domains/src/staking.rs#L2124

Reply:
The withdraw_share_amount is calculated from WithdrawParams.withdraws, not WithdrawParams.expected_withdraw.

There might be some way to rewrite the test to avoid it, but I couldn't work out how. Happy to leave a TODO?

```suggestion
                // TODO: work out how to avoid this multiplication on WithdrawParams.withdraws
                STORAGE_FEE_RESERVE
                    .left_from_one()
                    .mul_ceil(operator_stake)
```